batcher/aligned-batcher-lib/src/types/mod.rs

```diff
@@ -143,24 +148,35 @@ pub fn parse_proving_system(proving_system: &str) -> anyhow::Result<ProvingSyste
     }
 }
 
-#[cfg(test)]
-mod test {
-    use super::*;
-
-    #[test]
-    fn hash_new_parent_is_correct() {
-        let mut hasher = Keccak256::new();
-        hasher.update(vec![1u8]);
-        let child_1 = hasher.finalize_reset().into();
-        hasher.update(vec![2u8]);
-        let child_2 = hasher.finalize().into();
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ClientMessage {
+    pub verification_data: VerificationData,
+    pub signature: Signature,
+}
 
-        let parent = VerificationCommitmentBatch::hash_new_parent(&child_1, &child_2);
+impl ClientMessage {
+    /// Client message is a wrap around verification data and its signature.
+    /// The signature is obtained by calculating the commitments and then hashing them.
+    pub async fn new(verification_data: VerificationData, wallet: Wallet<SigningKey>) -> Self {
+        let hashed_leaf = VerificationCommitmentBatch::hash_data(&verification_data.clone().into());
+        let signature = wallet.sign_message(hashed_leaf).await.unwrap();
 
-        // This value is built using Openzeppelin's module for Merkle Trees, in particular using
-        // the SimpleMerkleTree. For more details see the openzeppelin_merkle_tree/merkle_tree.js script.
-        let expected_parent = "71d8979cbfae9b197a4fbcc7d387b1fae9560e2f284d30b4e90c80f6bc074f57";
+        ClientMessage {
+            verification_data,
+            signature,
+        }
+    }
 
-        assert_eq!(hex::encode(parent), expected_parent)
+    /// The signature of the message is verified, and when it correct, the
+    /// recovered address from the signature is returned.
+    pub fn verify_signature(&self) -> Result<Address, SignatureError> {
+        let hashed_leaf =
+            VerificationCommitmentBatch::hash_data(&self.verification_data.clone().into());
+        println!("HASHED LEAF: {:?}", hashed_leaf);
+        let recovered = self.signature.recover(hashed_leaf)?;
+        println!("SIGNATURE: {}", self.signature);
+        println!("RECOVERED ADDR: {:?}", recovered);
```

Comment:
Should probably remove logs, or at least move them to debug

Reply:
removed in [af418b8](https://github.com/yetanotherco/aligned_layer/pull/471/commits/af418b8f631c160a87235a40b5f5d97d66010f22) 🫡